src/outfit.ts

```diff
@@ -338,7 +338,8 @@ export class Outfit {
     }
     if (spec.bonuses) {
       for (const [item, value] of spec.bonuses) {
-        succeeded &&= value + this.getBonus(item) === this.addBonus(item, value);
+        this.addBonus(item, value);
+        succeeded &&= this.bonuses.has(item);
```

Comment:
Right now, this will still fail if the outfit cannot possibly equip the bonus item, because `.addBonus` calls `.setBonus` which calls `.canEquip`. Is this what we want to happen?

It might be smoother to remove the `.canEquip` call from `.setBonus`. Then the `.addBonus`/`.setBonus` call can never fail, and we rely on the maximizer to ignore impossible bonus items.